From 53aaac7ec74d628011f74ec74410145611cf731b Mon Sep 17 00:00:00 2001
From: Matteo Bertozzi <mbertozzi@apache.org>
Date: Wed, 24 Apr 2013 09:34:40 +0000
Subject: [PATCH 78/96] HBASE-8413 Snapshot verify region will always fail if the HFile has been archived

git-svn-id: https://svn.apache.org/repos/asf/hbase/branches/0.94@1471323 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 36e465b03b2e8443ed485e10f6a2c0ac1374db57)

Reason: Bug
Author: Jerry He
Ref: CDH-10583
---
 .../master/snapshot/MasterSnapshotVerifier.java    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/main/java/org/apache/hadoop/hbase/master/snapshot/MasterSnapshotVerifier.java b/src/main/java/org/apache/hadoop/hbase/master/snapshot/MasterSnapshotVerifier.java
index 4a4457e..db89e66 100644
--- a/src/main/java/org/apache/hadoop/hbase/master/snapshot/MasterSnapshotVerifier.java
+++ b/src/main/java/org/apache/hadoop/hbase/master/snapshot/MasterSnapshotVerifier.java
@@ -222,7 +222,7 @@ public final class MasterSnapshotVerifier {
         String fileName = hfile.getPath().getName();
         Path file = new Path(realCfDir, fileName);
         Path archived = new Path(archivedCfDir, fileName);
-        if (!fs.exists(file) && !file.equals(archived)) {
+        if (!fs.exists(file) && !fs.exists(archived)) {
           throw new CorruptedSnapshotException("Can't find hfile: " + hfile.getPath()
               + " in the real (" + realCfDir + ") or archive (" + archivedCfDir
               + ") directory for the primary table.", snapshot);
-- 
1.7.0.4

